### PR TITLE
Completes ncnews-9

### PR DESCRIPTION
**Pull request by garetheverson**
200 path & 404, 400 error paths: POST /api/articles/:article_id/comments

***ADDED***
app.post('/api/articles/:article_id/comments', postCommentByArticleId)
users.models - selectUserByUsername
articles.models - insertCommentByArticleId
articles.controllers - postCommentByArticleId, checking using selectUserByUsername(users.models) and insertCommentByArticleId (articles.models)

Tests:
Success path: POST /api/articles/:article_id/comments
    ✓ 201: Responds with the posted comment
Errors: POST requests for /api/articles/:article_id/comments
    ✓ 404: Responds with 'No article found' message given article_id that doesn't exist in POST
    ✓ 404: Responds with 'Author not found' message given username that doesn't exist in POST
    ✓ 404: Responds with 'Invalid path' given POST path that doesn't exist
    ✓ 400: Responds with 'Article ID must be a number' message given article_id that isn't a number in POST URL
    ✓ 400: Responds with 'Article ID must be a number' message given article_id that is a SQL injection in POST URL

***CHANGED***
No further changes

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -120,6 +120,79 @@ describe('Errors: GET requests for /api/articles/:article_id/comments', () => {
...
+    return request(app)
+      .post(`/api/articles/${identifier}/comments`)
+      .send(newComment)
+      .expect(400)
+      .then((res) => {
+        expect(res.body.msg).toBe('Article ID must be a number');
+      });
+  });
+});
+
```
Comment (karlaferry) [inline __tests__/app.test.js:195]: Good tests, well written. But how do you handle it if the post body is missing some/all fields or properties?

File: controllers/articles.controllers.js [hunk 2]
```diff
@@ -38,6 +41,27 @@ exports.getCommentsByArticleId = (req, res, next) => {
...
+    return next({ status: 400, msg: 'Article ID must be a number' });
+  }
+
+  selectArticleById(article_id)
+    .then((res) => {
+      return selectUserByUsername(username);
+    })
+    .then((res) => {
+      return insertCommentByArticleId(article_id, username, body);
+    })
```
Comment (karlaferry) [inline controllers/articles.controllers.js:57]: These two previous then blocks could be combined into one promise.all!

**Review by karlaferry — CHANGES_REQUESTED** [review]
Looks good. Just need one more error handler!

File: __tests__/app.test.js (hunk 1 above)
Comment (garetheverson) [reply __tests__/app.test.js:221]: Added in commit https://github.com/garetheverson/back-end-news-portfolio-project/pull/9/commits/f166b9bfa823b31d04499d6e48eb3f0abab83401 @karlaferry. Thanks for the feedback.

**Pull request merged**